pkg/buildkit/worker/pool.go

```diff
@@ -375,6 +375,7 @@ func (p *workerPool) updateWorkers(ctx context.Context) error {
 			if req := p.requests.Dequeue(); req != nil {
 				log.Info("Found pending pod request, processing")
 
+				pod := pod
```

Comment:
is this changing scope somehow?

Reply:
TL;DR: **Implicit memory aliasing**

https://husni.dev/beware-of-implicit-memory-aliasing-in-go-foor-loop/

I ran into this problem yesterday and, while I didn't see it happening here, this just safeguards against it.